src/redux/swap.ts

```diff
@@ -147,7 +150,8 @@ export const updateSwapInputCurrency = (
       type === ExchangeModalTypes.swap &&
       newInputCurrency?.type !== outputCurrency?.type &&
       newInputCurrency &&
-      !ignoreTypeCheck
+      !ignoreTypeCheck &&
+      !crosschainEnabled
```

Comment:
i think we want we want this specifically for cross chain swaps? instead this is disabling same chain swaps as well if crosschain swaps is enabled
we probably don't need it for now